addons/project/models/project_task.py

```diff
@@ -926,7 +926,8 @@ def create(self, vals_list):
                     project_id
                 ).company_id.id
             if not project_id and ("stage_id" in vals or self.env.context.get('default_stage_id')):
-                vals["stage_id"] = False
+                if not is_portal_user:
+                    vals["stage_id"] = False
```

Comment:
The stage won't be set if an administrator creates a task from a project sharing.